src/feature/nodelist/dirlist.c

```diff
@@ -49,6 +49,31 @@ static smartlist_t *trusted_dir_servers = NULL;
  * and all fallback directory servers. */
 static smartlist_t *fallback_dir_servers = NULL;
 
+/** Return true iff the given v4 address is the one of a configured directory
+ * authority matching the given type.
+ *
+ * This needs to be fast because it can be called at each directory connection
+ * in order to learn if the request is from an authority or not. */
+bool
+dirlist_addr_is_trusted_dir_by_type(const tor_addr_t *addr,
```

Comment:
The documentation for this function should emphasize the point that we should never actually trust this function to mean that we have a real directory authority, since it is just an address check, and IP spoofing is possible.